src/lib/es2015.collection.d.ts

```diff
@@ -42,7 +42,13 @@ interface ReadonlyMap<K, V> {
     readonly size: number;
 }
 
-interface WeakMap<K extends object, V> {
+type WeakKeyTypes = WeakKeyTypesStore[keyof WeakKeyTypesStore];
+
+interface WeakKeyTypesStore {
+    object: object;
+}

```

Comment:
Ron suggests we use a shorter name of `WeakKeyTypes`: https://github.com/microsoft/TypeScript/issues/52534#issuecomment-1428822049

Small edit: I suggest we also move the `WeakKeyTypes` below `WeakKeyTypesStore`, rather than referencing it before it's defined. Even though it works, feels slightly easier to read the other way around.

```suggestion
interface WeakKeyTypes {
    object: object;
}
type WeakKey = WeakKeyTypes[keyof WeakKeyTypes];
```

Maybe we could also add a small comment explaining the purpose of this type. Similar to how there is a comment for the similar concept being used for ArrayBuffer and SharedArrayBuffer here: https://github.com/microsoft/TypeScript/blob/f64f40d20511d2d2a9eb4fcc1ad6750977be2d40/src/lib/es5.d.ts#L1655-L1661 What do you think?

Reply:
That seems more than reasonable to me, I will add a short comment!